### ISSUE TITLE
Unexpected content getting parsed by Stylelint

### ISSUE THREAD

**Issue by spmcbride1201** [styled-components/stylelint-processor-styled-components#153]
Team Stylelint:

Love your ecosystem! I've been trying to add more tooling to help our junior devs, and I've been having some issues getting Stylelint to behave.

When I run `stylelint 'src/**/*.{js,jsx}'`, I'm currently getting errors. It seems that stylelint is trying to parse code outside of my styled-component template strings. 

```bash
src/components/Button/components/ButtonWrap.js
 75:21  ✖  Unknown word   CssSyntaxError

src/components/AppHeader/components/Tab/components/TabLink.js
 25:23  ✖  Missed semicolon   CssSyntaxError
```
## ButtonWrap
![image](https://user-images.githubusercontent.com/8700322/33858245-eb9a3f44-de9c-11e7-8bdd-60748af623a0.png)

## TabLink

![image](https://user-images.githubusercontent.com/8700322/33859841-d23a5320-dea3-11e7-8d17-a5fe79a43604.png)

```js
import { NavLink } from "react-router-dom";
import styled from "styled-components";

import { spacingScale, contrastColor } from "style/styleFunctions";
import {
  COLOR_ALT_BACKGROUND,
  COLOR_WHITE,
  COLOR_HIGHLIGHT
} from "style/styleVariables";

const TAB_WIDTH_BASE = "1%";
const COLOR_TAB_BACKGROUND_BASE = contrastColor(COLOR_ALT_BACKGROUND, 0.175);
const COLOR_TAB_BACKGROUND_ACTIVE = contrastColor(COLOR_ALT_BACKGROUND, 0.3);
const COLOR_TAB_HIGHLIGHT = COLOR_HIGHLIGHT;
const TabLink = styled(NavLink).attrs({
  exact: true
})`
  border-top: 1px solid transparent;
  font-weight: 600;
  flex: 1 1 ${TAB_WIDTH_BASE};
  margin: 1px 1px 0 0;
  padding: ${spacingScale(0.5)} ${spacingScale(1)} ${spacingScale(1)};
  position: relative;
  transition: all 0.15s ease; 
  &:after {
    border-radius: 1px;
    box-shadow: 0 0 1em 1px ${COLOR_TAB_HIGHLIGHT.string()};
    color: inherit;
    content: '';
    height: 1px;
    left: 0;
    opacity: 0;
    pointer-events: none;
    position: absolute;
    right: 0;
    top: -2px;
    z-index: 100;
  }

  &:hover {
    background-color: ${/* sc-block */ COLOR_TAB_BACKGROUND_BASE.mix(
      COLOR_TAB_BACKGROUND_ACTIVE,
      0.3
    ).string()};

    &:after {
      opacity: .5;
    }
  }

  &.active {
    background-color: ${COLOR_TAB_BACKGROUND_ACTIVE.string()};
    &:after {
      background-color: ${COLOR_TAB_HIGHLIGHT.mix(COLOR_WHITE, 0.3).string()};
      opacity: 1;
      transition: .1s;
    }
  }

  &:active {
    &:after {
      background-color: ${COLOR_TAB_HIGHLIGHT.mix(COLOR_WHITE, 0.3).string()};
      opacity 1;
      transition: .1s;
    }
  }
  
  &:focus {
    outline: ${COLOR_TAB_HIGHLIGHT.string()} auto 3px;
  }
`;

export default TabLink;
```

## `.stylelintrc` file at root of project
```json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "syntax": "scss",
  "plugins": [
    "stylelint-no-unsupported-browser-features"
  ],
  "rules": {
    "plugin/no-unsupported-browser-features": [true, {
      "ignore": ["rem"],
      "severity": "warning"
    }],
    "at-rule-empty-line-before": null,
    "block-opening-brace-space-after": null,
    "block-closing-brace-space-before": null,
    "color-hex-length": null,
    "comment-empty-line-before": null,
    "declaration-block-semicolon-newline-after": null,
    "declaration-colon-newline-after": null,
    "declaration-empty-line-before": null,
    "indentation": null,
    "no-extra-semicolons": null,
    "rule-empty-line-before": null,
    "selector-pseudo-element-colon-notation": null,
    "value-list-max-empty-lines": null,
    "value-list-comma-newline-after": null
  }
}
```

**Comment by emilgoldsmith** [comment]
Hey! 

Just to rule out the obvious first, are you using the latest version of the processor? Since we recently published a fix that was causing multiline interpolations to break the sourcemaps, and therefore report errors in wrong places?

**Comment by spmcbride1201** [comment]
Updated to `1.2.1` and updated appropriately. One of the errors now seems to be an erroneous "Missing semicolon" error, but that's probably now a `stylelint` thing. ButtonWrap seems to behave the same.

**Comment by emilgoldsmith** [comment]
Okay, I'll try and run it locally and see if I can debug it a bit, I'll be back!

**Comment by emilgoldsmith** [comment]
For one I just noticed that you have written `&.active` instead of `&:active` here:
```css
&.active {
    background-color: ${COLOR_TAB_BACKGROUND_ACTIVE.string()};
    &:after {
      background-color: ${COLOR_TAB_HIGHLIGHT.mix(COLOR_WHITE, 0.3).string()};
      opacity: 1;
      transition: .1s;
    }
  }
```

**Comment by emilgoldsmith** [comment]
Okay I think I identified at least the error in TabLink, it's an error introduced by the new fixes I mentioned previously, and it might cover both your cases. I'll see if I can get around to it tonight, but yep, definitely our bad. Thanks for the report!

**Comment by emilgoldsmith** [comment]
Hey, a bit more information. Firstly, to fix your css you currently have 2 bugs in TabLink:

1. You're using the wrong interpolation tag in your multiline interpolation, it should be `/* sc-value */` not `/* sc-block */`. Check out [this vocabulary list] if it wasn't just a typo and you want to see where you went wrong.

2. In the second to last block you have forgotten a colon in your opacity statement.

This should make your code run.

As you can see in the issue I just referenced this one from there's apparently a problem with running source maps when a CssSyntaxError occurs, which sucks! Hope we can find a way around that. I also think I might've spotted another bug which is what I was talking about above, but in the TabLink case it seems that it didn't affect you.

**Comment by emilgoldsmith** [comment]
If you would like me to take a look at ButtonWrap I'll need the whole file since the sourcemaps aren't working, so the problem is probably simply in your CSS (though it could also possibly be some issues with our implementation of interpolations) but just in a completely different place, probably in the 75th line of CSS present in your file (skipping all non css stuff)

**Comment by spmcbride1201** [comment]
Emil, thank you so much for your keen eye! TabLink is behaving again, and now that I understand that the syntax error is related to the non-sourcemapped CSS string, I think I should be able to track things down in ButtonWrap. This file is essentially a big ball of styles and mixins that we refactored from SCSS earlier this week, so it might take a bit. I'll report back when I've tracked down the culprit.

**Comment by spmcbride1201** [comment]
After spending a few hours refactoring things, I've found and fixed a handful of syntax errors. However, I still am getting the error at approximately the same line number as before, so I likely missed the major one (not surprising, as I am a quite a CSS amateur) . Since I see that you're working on submitting a PR upstream, I'm thinking I'll stand down until that's worked it's way through the ecosystem. I'm happy to either test your candidate fix or provide you the full breaking code if you think that will help.

**Comment by emilgoldsmith** [comment]
Hey! If you want you can replace the `lib` folder in your `node_modules/stylelint` folder with the one from here: https://github.com/emilgoldsmith/stylelint.

I believe I have it working now, just gonna finish up tests and gotta go through code review etc.

**Comment by emilgoldsmith** [comment]
Oh sorry, forgot I hadn't pushed + wasn't linking to the branch (that didn't exist then either). Here's the correct link: https://github.com/emilgoldsmith/stylelint/tree/feature/apply-resultprocessors-on-CssSyntaxError

**Comment by spmcbride1201** [comment]
That seemed to do it! I went to the line and column suggested and discovered a malformed comment. Fixing it seems to have resolved things, so I think you're on the right track.

The only oddity I encountered was that your default formatter seemed to be json for some reason, so I had to change my npm script to manually set it back to to string via `"lint-css": "stylelint --formatter string 'src/**/*.{js,jsx}'",`

You can feel free to close out this comment if you want. I'll keep track of your progress in your other issue.

Thanks so much!

**Comment by emilgoldsmith** [comment]
> The only oddity I encountered was that your default formatter seemed to be json for some reason

Didn't change anything there so that sounds weird, I'm pretty sure the problem should be at your end. 

And yep I'll close this one

**Comment by emilgoldsmith** [comment]
We just published v1.3.0 that together with stylelint v9+ should now apply sourcemaps correctly to css syntax errors!
